### ISSUE TITLE
Support connecting to a WPA3 AP from ESP32 support

### ISSUE THREAD

**Issue by averyanalex** [esp-rs/esp-hal#1600]
What about WPA3Personal support? It's [already supported](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-guides/wifi-security.html) in esp-idf.

**Comment by ada-magicat** [comment]
> This is not the first time I come back to this issue to see if it has some useful info for me but then leave disappointed. Could we please edit the title to indicate that this is about an ESP STA trying to connect to a WPA3 access point, and not necessarily WPA3 support in softAP?

What useful information are you looking for? I can reproduce this on two different WPA3 networks with a minimal program that just tries to connect to the access point on my esp32c6. Using `AuthMethod::Wpa3Personal` or `AuthMethod::Wpa2Wpa3Personal` seems to make no difference.

**Comment by andrico21** [comment]
3 years have passed since this thread is started, and there are still no useful updates - does nobody use WPA3, or does the issue really seem that insignificant? is it a bug (so, "_dear community, please help us with logs_"), is it a fundamental ESP32 issue probably (like "_don't even try guys - it won't work on this HW, wait for next gen microcontrollers_")...

**no offense at all, really** guys but just wondering - is it really considered so insignificant? It's like 8 years since WPA3 was released - consumer routers from 2020+ ship with WPA3 defaults, and some (especially mesh systems) offer WPA3-only mode, `no_std` ESP devices simply cannot connect.

The underlying binary blobs already contain SAE support (since used successfully via `esp-idf-svc`) - isn't it?

**Comment by bjoernQ** [comment]
> does nobody use WPA3, or does the issue really seem that insignificant?

We found that most environments are at least configured for WPA2+WPA3 - which makes this feature less urgent. Also WPA2 vs WPA3 is (other than WEP vs WPA) doesn't impose a significant risk. Given that, the priority of this feature request is lower compared to other tasks

> is it a bug (so, "dear community, please help us with logs"), is it a fundamental ESP32 issue probably (like "don't even try guys - it won't work on this HW, wait for next gen microcontrollers")... 

It's a missing feature - not a bug - the HW is ofc not a limiting factor here

> The underlying binary blobs already contain SAE support (since used successfully via esp-idf-svc) - isn't it?

Partially, yes — but enabling WPA3 in this stack requires enabling mbedTLS for the supplicant. That’s not a small change: it has noticeable footprint and integration implications, and currently conflicts with EAP-FAST support. So it’s more involved than simply toggling an existing capability.

TL;DR: WPA3 support is on the roadmap, but not a high-priority item right now due to trade-offs and existing workload.

That said, contributions are very welcome - if you’re interested in working on this, we’d be happy to review a PR or help clarify implementation details.

**Comment by andrico21** [comment]
> Partially, yes — but enabling WPA3 in this stack requires enabling mbedTLS for the supplicant. That’s not a small change: it has noticeable footprint and integration implications, and currently conflicts with EAP-FAST support. So it’s more involved than simply toggling an existing capability.
> 
> TL;DR: WPA3 support is on the roadmap, but not a high-priority item right now due to trade-offs and existing workload.

@bjoernQ thanks a lot for detailed reply, appreciate it so much!

**Comment by ada-magicat** [comment]
@bjoernQ Appreciate the reply. I understand it being low priority, but if it is a missing feature then it should **not** be exposed in the API. The fact that `AuthMethod::Wpa3Personal` is in the API, is not documented as being non-functional and the fact that I can use it in my code, my program compiles and tries to run when the underlying support is not there is a bug IMHO. This needs a documentation note, the temporary removal of those flags or to cause a compilation error when trying to use it. I understand it will take time and work to support this, so I think it should not be exposed until that is done to clearly communicate what is supported by this software stack.
